src/wmtk/multimesh/utils/extract_child_mesh_from_tag.cpp

```diff
@@ -0,0 +1,138 @@
+#include "extract_child_mesh_from_tag.hpp"
+#include <map>
+#include <wmtk/EdgeMesh.hpp>
+#include <wmtk/Mesh.hpp>
+#include <wmtk/MultiMeshManager.hpp>
+#include <wmtk/PointMesh.hpp>
+#include <wmtk/Primitive.hpp>
+#include <wmtk/TetMesh.hpp>
+#include <wmtk/TriMesh.hpp>
+
+
+namespace wmtk::multimesh::utils {
+
+std::shared_ptr<Mesh> extract_and_register_child_mesh_from_tag(
+    Mesh& m,
+    const std::string& tag,
+    const long tag_value,
+    const PrimitiveType& pt)
+{
+    assert(m.top_simplex_type() >= pt);
+    auto tag_handle = m.get_attribute_handle<long>(tag, pt);
+    return extract_and_register_child_mesh_from_tag_handle(m, tag_handle, tag_value, pt);
+}
+
+
+std::shared_ptr<Mesh> extract_and_register_child_mesh_from_tag_handle(
+    Mesh& m,
+    const MeshAttributeHandle<long>& tag_handle,
+    const long tag_value,
+    const PrimitiveType& pt)
+{
+    auto tags = m.create_const_accessor(tag_handle);
+
+    std::vector<Tuple> tagged_tuples;
+    for (auto t : m.get_all(pt)) {
+        if (tags.const_scalar_attribute(t) == tag_value) {
+            tagged_tuples.emplace_back(t);
+        }
+    }
+
+    switch (pt) {
+    case PrimitiveType::Vertex: throw("not implemented");
+    case PrimitiveType::Edge: {
+        std::map<long, long> parent_to_child_vertex_map;
+        long child_vertex_count = 0;
+
+        RowVectors2l edge_mesh_matrix;
+        edge_mesh_matrix.resize(tagged_tuples.size(), 2);
+
+        for (long i = 0; i < tagged_tuples.size(); ++i) {
+            const long v1 = m.id(tagged_tuples[i], PrimitiveType::Vertex);
```

Comment:
make this a `const std::array<long,2>`. this allows for line 55 and 59 to turn into a for loop, same with 65/66.